### DISCUSSION TITLE
LapTimes

### DISCUSSION THREAD

**Discussion by hash55** [theOehrly/Fast-F1#101] [Q&A]
Hi! Is it possible to get laptimes before 2018? As far as I'm concerned ergast provides laptimes since 2011 and on the fia web there are laptimes available even from 2009. However, I don't know how check this with fastf1. 
Thanks

**Comment by theOehrly** [comment]
No, this is not possible at the moment. FastF1 currently utilizes only a small subset of the data that Ergast can provide. The main incentive currently is to access the "official" F1 data. Ergast is only used to fill in some gaps.

I see how it would be nice to have the old data available in FastF1 as well. The main issue here is the fact that this introduces some inconsistencies. Ergast provides substantially less information. Therefore, the Laps data object will be missing plenty of data. I'd have to check how many problems this actually creates. It might also be somewhat confusing for a user if functionality exists for some seasons and not for others. Proper error handling would need to be implemented so that the actual source of the problem is communicated to the user.

Maybe I can add this functionality. But I'll have to see. There are (in my opinion) more important issues/features that need to be looked at first.

Do you know if there are any other python modules that provide this functionality?

**Comment by hash55** [comment]
Hi! 
I understand that there are more important features than this one (such as the sample rate), so take it easy.
What I found was  [this](https://github.com/jshumphrey/f1db) repo, which provides some ergast data in sql. (laptimes, overtakes etc)
Don't worry at all if you don't have time to look at this, I was just curious about data of older races.

**Reply by theOehrly** [reply]
Interesting, thanks for digging this up. I'll take a look at it when I've got some time.
